diff --git a/cacti/nuca.cc b/cacti/nuca.cc
index 230025c..5d615ff 100644
--- a/cacti/nuca.cc
+++ b/cacti/nuca.cc
@@ -43,7 +43,7 @@ unsigned int MIN_BANKSIZE=65536;
 int cont_stats[2 /*l2 or l3*/][5/* cores */][ROUTER_TYPES][7 /*banks*/][8 /* cycle time */];
 
   Nuca::Nuca(
-      TechnologyParameter::DeviceType *dt = &(g_tp.peri_global)
+      TechnologyParameter::DeviceType *dt
       ):deviceType(dt)
 {
   init_cont();
diff --git a/cacti/nuca.h b/cacti/nuca.h
index a00044e..ce27ae7 100644
--- a/cacti/nuca.h
+++ b/cacti/nuca.h
@@ -76,7 +76,7 @@ class Nuca : public Component
 {
   public:
     Nuca(
-        TechnologyParameter::DeviceType *dt);
+        TechnologyParameter::DeviceType *dt = &(g_tp.peri_global));
     void print_router();
     ~Nuca();
     void sim_nuca();
